plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceRuntimeInfo.java

```diff
@@ -234,124 +229,20 @@ public String projectsRoot() {
 
     @Override
     public Map<String, ServerImpl> getServers() {
+        ServerEvaluationStrategy strategy = provider.getStrategy(info, serversConf, internalHost);
```

Comment:
I think that I imaged this code in a wrong way. Because it looks just like I described and it looks not very good. 
What do you think if:
1. Strategies are injected in MapBinder
2. Some strategy provider is bound as provider of Server**s**EvaluationStrategy
3. In constructor Server**s**EvaluationStrategy is injected
4. In constructor strategy is assigned to a field.
5. In getServers we call strategy.getServers(info, internalHost) and it returns Map<String, ServerImpl> with filled ports and addresses.

@l0rd @amisevsk WDYT?